pkg/aws/snow.go

```diff
@@ -0,0 +1,56 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+)
+
+const (
+	snowEC2Port = 8243
+)
+
+func snowEndpoint(deviceIP string) *ServiceEndpoint {
+	return &ServiceEndpoint{
+		ServiceID:     "EC2",
+		SigningRegion: "snow",
+		URL:           fmt.Sprintf("https://%s:%d", deviceIP, snowEC2Port),
+	}
+}
+
+func WithCustomCABundleFile(certsFile string) config.LoadOptionsFunc {
+	caPEM, err := ioutil.ReadFile(certsFile)
+	if err != nil {
+		return func(*config.LoadOptions) error {
+			return fmt.Errorf("reading aws certificates file: %v", err)
+		}
+	}
+
+	return config.WithCustomCABundle(bytes.NewReader(caPEM))
```

Comment:
This is OK but there is a nuance in behavior with respect to what a maintainer intuitively expects (probably).

Intuitively I feel a maintainer, given the `WithCustomCABundleFile` doesn't return an error, would assume the file is read _when the option is applied_. That gives cause for me to, hypothetically, write a file _after_ calling `WithCustomCABundleFile` but _before_ the option is applied. While that may seem silly, its quite legitimate when reflecting on the package API and would result in odd behavior because the option would always error.

Below is a suggestion to defer reading the file and still apply the option. It just constructs the actual option in-line when its needed. I also took the liberty to swap out reading the entire file into memory with a buffered io stream to be a little more efficient (probably negligable given the pem won't be huge but its good practice to prefer streams nonetheless; the buffered reader is likely unnecessary too but doesn't hurt). Finally, I changed the error to include the root error as `%w` meaning consumers can get at the root cause if they desire (its a standard library error so doesn't hurt making it part of the pkg api).

```suggestion
	return func(opts *config.LoadOptions) error {
		caPEM, err := os.Open(certsFile)
		if err != nil {
			return fmt.Errorf("reading aws certificates file: %w", err)
		}
		
		customBundleOpt := config.WithCustomCABundle(bufio.NewReader(caPEM))
		if err := customBundleOpt(opts); err != nil {
			return err
		}
		
		return 
	}
```